atlasdb-calcite/src/main/java/com/palantir/atlasdb/calcite/AtlasColumnMetdata.java

```diff
@@ -0,0 +1,201 @@
+            return column().get().getLongName();
+        } else if (isValue()) {
+            return "val";
+        }
+        throw new IllegalStateException();
+    }
+
+    @Value.Derived
+    public String getName() {
+        if (column().isPresent()) {
+            return column().get().getShortName();
+        }
+        return getDisplayName();
+    }
+
+    public RelDataType relDataType(RelDataTypeFactory factory) {
+        RelDataType type = valueTypeToDataType(factory);
+        if (isComponent() || type.isNullable()) {
+            return type;
+        }
+        return factory.createTypeWithNullability(type, true);
+    }
+
+    private RelDataType valueTypeToDataType(RelDataTypeFactory factory) {
+        switch (format()) {
+            case PROTO:
+                return factory.createMapType(
+                        factory.createJavaType(String.class),
+                        factory.createSqlType(SqlTypeName.ANY));
+            case PERSISTABLE:
+            case PERSISTER:
+                return factory.createSqlType(SqlTypeName.BINARY);
+            case VALUE_TYPE:
+                switch (valueType()) {
+                    case VAR_LONG:
+                    case VAR_SIGNED_LONG:
+                    case FIXED_LONG:
+                    case FIXED_LONG_LITTLE_ENDIAN:
+                        return factory.createJavaType(Long.class);
+                    case NULLABLE_FIXED_LONG:
+                        return factory.createTypeWithNullability(factory.createJavaType(Long.class), true);
+                    case VAR_STRING:
+                    case STRING:
+                        // for now we choose to represent ValueType blobs as strings
+                    case BLOB:
+                    case SIZED_BLOB:
+                        return factory.createJavaType(String.class);
+                    case SHA256HASH:
+                        return factory.createJavaType(Sha256Hash.class);
+                    case UUID:
+                        return factory.createJavaType(UUID.class);
+                    default:
+                        throw new IllegalStateException("Unknown ValueType: " + valueType());
+                }
+            default:
+                throw new IllegalStateException("Unknown Format: " + format());
+        }
+    }
+
+    public Object deserialize(byte[] bytes) {
+        if (isComponent()) {
+            throw new RuntimeException("Row/column components cannot be deserialized in isolation."
+                    + "Use EncodingUtils.fromBytes() instead.");
+        }
+        switch (format()) {
+            case PROTO:
+                if (isComponent()) {
+                    throw new IllegalStateException("Row/column components cannot contain protobufs (invalid metadata).");
+                } else if (isNamedColumn()) {
+                    return ProtobufDeserializers.convertMessageToMap(
+                            column().get().getValue().hydrateProto(
```

Comment:
@gbonik pointed out that hydrateProto() requires the object class file, but convertMessageToMap does not use the class information.  It is definitely worth just loading the protobuf message directly without requiring the class file.